src/Features/Core/Portable/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -368,179 +390,73 @@ internal abstract partial class AbstractInheritanceMarginService
                 // then don't show the derived overriden & implemented method in the default implementation for System.Threading.SynchronizationContext in metadata
                 var overridingSymbols = allOverridingSymbols.WhereAsArray(symbol => symbol.Locations.Any(static l => l.IsInSource));
 
-                if (overridingSymbols.Any() || overriddenSymbols.Any() || implementedSymbols.Any())
+                var targetsForOverriddenSymbols = await CreateInheritanceTargetsAsync(
+                    solution,
+                    overriddenSymbols,
+                    InheritanceRelationship.OverriddenMember,
+                    cancellationToken).ConfigureAwait(false);
+
+                var targetsForImplementedSymbols = await CreateInheritanceTargetsAsync(
+                    solution,
+                    implementedSymbols,
+                    InheritanceRelationship.ImplementedMember,
+                    cancellationToken).ConfigureAwait(false);
+
+                var targetsForOverridingSymbols = await CreateInheritanceTargetsAsync(
+                    solution,
+                    overridingSymbols,
+                    InheritanceRelationship.OverridingMember,
+                    cancellationToken).ConfigureAwait(false);
+
+                if (!targetsForOverriddenSymbols.IsEmpty || !targetsForImplementedSymbols.IsEmpty || !targetsForOverridingSymbols.IsEmpty)
                 {
-                    var item = await CreateInheritanceMemberItemForClassOrStructMemberAsync(solution,
+                    var item = CreateInheritanceMarginItem(
                         memberSymbol,
                         lineNumber,
-                        implementedMembers: implementedSymbols,
-                        overridingMembers: overridingSymbols,
-                        overriddenMembers: overriddenSymbols,
-                        cancellationToken).ConfigureAwait(false);
+                        targetsForOverriddenSymbols.Concat(targetsForImplementedSymbols).Concat(targetsForOverridingSymbols));
                     builder.AddIfNotNull(item);
```

Comment:
sice you're already doing 'AddIfNotNull' can you just make CreateInheritanceMarginItem return null if it is given no targets?  then you don't need to do the `if` check here, and can move into a common location (that also applies to the several cases above as well).

Reply:
Same as below